Bug: rapina new generates project missing hyper dependency

## Description

When creating a new project with `rapina new my-app`, the generated code doesn't compile because `hyper` is missing from the dependencies.

## Steps to Reproduce

```bash
rapina new my-app
cd my-app
cargo build
```

## Error

```
error[E0433]: failed to resolve: use of undeclared crate or package `hyper`
  --> src/main.rs:...
   |
   |     _req: hyper::Request<hyper::body::Incoming>,
   |           ^^^^^ use of undeclared crate or package `hyper`
```

## Root Cause

The generated `src/main.rs` uses `hyper` types directly:
```rust
async fn hello(
    _req: hyper::Request<hyper::body::Incoming>,
    _params: rapina::extract::PathParams,
    _state: std::sync::Arc<rapina::state::AppState>,
) -> &'static str {
    "Hello from Rapina!"
}
```

But the generated `Cargo.toml` doesn't include `hyper`:
```toml
[dependencies]
rapina = "0.1"
tokio = { version = "1", features = ["full"] }
serde = { version = "1", features = ["derive"] }
serde_json = "1"
```

## Solution Options

1. **Add hyper to generated Cargo.toml** - Quick fix but adds explicit dependency
2. **Re-export hyper types from rapina** - Better DX, users don't need to know about hyper
3. **Simplify handler signature** - Use extractors instead of raw hyper types

Option 2 or 3 would provide better ergonomics for users.

## Labels
bug, cli

--- a/rapina-cli/src/commands/dev.rs
+++ b/rapina-cli/src/commands/dev.rs
@@ -1,11 +1,11 @@
 //! Implementation of the `rapina dev` command.
 
 use colored::Colorize;
-use notify_debouncer_mini::{new_debouncer, notify::RecursiveMode, DebounceEventResult};
+use notify_debouncer_mini::{DebounceEventResult, new_debouncer, notify::RecursiveMode};
 use std::path::Path;
 use std::process::{Child, Command, Stdio};
 use std::sync::atomic::{AtomicBool, Ordering};
-use std::sync::{mpsc, Arc};
+use std::sync::{Arc, mpsc};
 use std::time::Duration;
 
 /// Configuration for the dev server.
@@ -292,7 +292,8 @@ fn print_banner(config: &DevConfig) {
     println!();
     println!(
         " {}",
-        "╭───────────────────────────────────────────────────────────╮".custom_color(colors::mauve())
+        "╭───────────────────────────────────────────────────────────╮"
+            .custom_color(colors::mauve())
     );
 
     // Title - centered in 59 chars
@@ -307,7 +308,8 @@ fn print_banner(config: &DevConfig) {
 
     println!(
         " {}",
-        "├───────────────────────────────────────────────────────────┤".custom_color(colors::mauve())
+        "├───────────────────────────────────────────────────────────┤"
+            .custom_color(colors::mauve())
     );
 
     // Empty line
@@ -338,8 +340,7 @@ fn print_banner(config: &DevConfig) {
     println!(
         " {}{}{}",
         b,
-        format!(" {:<58}", "For production: cargo build --release")
-            .custom_color(colors::subtext()),
+        format!(" {:<58}", "For production: cargo build --release").custom_color(colors::subtext()),
         b
     );
 
@@ -348,7 +349,8 @@ fn print_banner(config: &DevConfig) {
 
     println!(
         " {}",
-        "╰───────────────────────────────────────────────────────────╯".custom_color(colors::mauve())
+        "╰───────────────────────────────────────────────────────────╯"
+            .custom_color(colors::mauve())
     );
     println!();
 }

--- a/rapina-cli/src/commands/new.rs
+++ b/rapina-cli/src/commands/new.rs
@@ -109,6 +109,7 @@ hyper = "1"
 /// Generate the content for src/main.rs.
 fn generate_main_rs() -> String {
     r#"use rapina::prelude::*;
+use rapina::middleware::RequestLogMiddleware;
 use serde::Serialize;
 use std::sync::Arc;
 
@@ -129,9 +130,9 @@ async fn main() -> std::io::Result<()> {
         .get_named("/", "hello", hello)
         .get_named("/health", "health_check", health);
 
-    println!("Starting Rapina server on http://127.0.0.1:3000");
-
     Rapina::new()
+        .with_tracing(TracingConfig::new())
+        .middleware(RequestLogMiddleware::new())
         .router(router)
         .listen("127.0.0.1:3000")
         .await

